app/controllers/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+class UsersController < ApplicationController
+  def new
+    if session[:name].nil?
+      @user = User.new
+    else
+      @user = User.find(session[:id])
+      flash[:info] = 'You are already signed in'
+      redirect_to @user
+    end
+  end
+
+  def index
+    @user = User.all
+  end
+
+  def show
+    @user = User.find_by(params[:id])
+    @created_events = @user.created_events
+    @upcoming_events = @user.upcoming_events
+    @previous_events = @user.previous_events
+  end
+
+  def create
+    @user = User.new(user_params)
+    @user.save
+    session[:name] = @user.name
+    session[:id] = @user.id
+    # cookies[:name] = @user.name
+    # cookies[:id] = @user.id
```

Comment:
You ought to remove commented out code.
It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history.
Ass you can see from [this screesnhot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from pathwright. This is a requirement to address all occurrences of commented out code.